gdal/frmts/netcdf/netcdfdataset.cpp

```diff
@@ -3392,6 +3403,11 @@ void netCDFDataset::SetProjectionFromVar( int nGroupId, int nVarId,
         // variables without same name than dimension using the same resolving
         // logic. This should handle for example NASA Ocean Color L2 products.
 
+        const bool bIgnoreXYAxisNameChecks =
+            CPLTestBool(CPLGetConfigOption("GDAL_NETCDF_IGNORE_XY_AXIS_NAME_CHECKS", "NO")) ||
+            // Dataset from https://github.com/OSGeo/gdal/issues/4075 has a res and transform attributes
+            (FetchAttr(nGroupId, nVarId, "res") != nullptr &&
```

Comment:
I wouldn't worry about using the res/transform attributes here either. It hasn't worked in the past and doesn't need to moving forward. That is the way `xarray` currently does it for metadata purposes only and will likely not be used moving forward [ref](https://github.com/pydata/xarray/issues/4697). `rioxarray` uses the method currently supported by GDAL with the `GeoTransform` attribute [ref](https://github.com/corteva/rioxarray/blob/c5203665aa1a7e76e4729df9a80568f999db01dc/rioxarray/rioxarray.py#L494-L496).

Reply:
Should also note that it isn't a GDAL transform: http://xarray.pydata.org/en/stable/generated/xarray.open_rasterio.html

Reply:
See the example in the docstring in the link above.